src/EnergyPlus/SolarShading.cc

```diff
@@ -5970,7 +5970,24 @@ namespace SolarShading {
 					// The inside reveals receive solar (reflected part + absorbed part) from the window, this amount should be
 					// deducted from the BTOTZone, then adds the InsRevealDiffIntoZone
 					if ( SurfaceWindow( SurfNum ).WindowModelType == WindowBSDFModel ) { //Complex Fenestration
-						SurfSolIncPtr = SurfaceScheduledSolarInc( BackSurfaceNumber, ConstrNumBack );
+						// Jason Glazer added the following code and note on March 16, 2015 to address Github #4777
+						//
+						// Originally this just had the line:
+						//
+						//   SurfSolIncPtr = SurfaceScheduledSolarInc( BackSurfaceNumber, ConstrNumBack );
+						//
+						// prior to the line: if ( SurfSolIncPtr == 0 ) {
+						//
+						// But since BackSurfaceNumber, ConstrNumBack have not been set, added the code to scan through the
+						// back surfaces to look for the value. Since there can be more than one back surface for each, just
+						// loop through it and use the final value.
+						SurfSolIncPtr = 0;
+						int NBkSurf = ShadowComb( SurfNum ).NumBackSurf;
+						for ( int IBack = 1; IBack <= NBkSurf; ++IBack ) {
+							BackSurfaceNumber = ShadowComb( SurfNum ).BackSurf( IBack );
+							ConstrNumBack = Surface( BackSurfaceNumber ).Construction;
+							SurfSolIncPtr = SurfaceScheduledSolarInc( BackSurfaceNumber, ConstrNumBack );
```

Comment:
Yes, I would agree that it should be only

SurfSolIncPtr = SurfaceScheduledSolarInc( SurfNum, ConstrNum );

This looks like copy/paste error. Idea of looking for this pointer is to determine if schedule file does exists and if file do exists then BTOTZone (total into zone) should not be calculated here at all but added from schedule file in some other place. If this line is removed that BTOTZone will be calculated here and then also later added from schedule file.

I would still need to check this claim.

However, I have found another potential problem:
I am running tests since yesterday. Currently performing huge one with Scheduled Surface Gains and I am getting warnings which does not seems good.

   ** Warning ** ProcessScheduleInput: Schedule:File="NCORE_LIGHT_SCH" 8760 records had errors - these values are set to 0.
   **   ~~~   ** Use Output:Diagnostics,DisplayExtraWarnings; to see individual records in error.

I am trying to compare results of EnergyPlus 8.2.0 (installed version) with this branch and I am getting this warning in both versions. I will try to run this in version 8.1 (latest fortran version) to confirm that message was not presented at a time there.